docker/templates/proxy_headers.conf.template

```diff
@@ -5,9 +5,9 @@ proxy_set_header X-Forwarded-Port $server_port;
 proxy_set_header X-Forwarded-Host $host;
 
 # Defaults to 60 seconds.
-#proxy_connect_timeout       300;
-#proxy_send_timeout          300;
-#proxy_read_timeout          300;
+proxy_connect_timeout       30;
+proxy_send_timeout          30;
+proxy_read_timeout          30;
```

Comment:
Is this going to mess with the cascading timeouts that are needed for the LB? The LB has 55sec and keepalive timeout is 60, and the direction of that difference was important iirc. ("make sure that the keep-alive timeout is greater than the idle timeout settings of your load balancer." [here](https://docs.aws.amazon.com/elasticloadbalancing/latest/classic/ts-elb-error-message.html#ts-elb-errorcodes-http503)).

I don't know how the proxy timeout is applied, so feel free to ignore me if this is a non issue. Just wanted to flag the idea just in case.

Reply:
To be honest, I'm not sure, but I had convinced myself that this just needed to be less than the warehouse lb timeout and would "kick out" connections that were stalled more quickly. With nginx doing buffering, I think these values just need to be lower than 65s.

I'll take your advice and put this back since I really don't know for sure, and we'll see if the more restrictive lifetimes of keepalive requests helps by itself. It's a better test to try one thing at a time anyway.

Reply:
The keepalive_timeout is the value that directly correlates with the load balancer idle timeout. Proxy timeouts may contribute to upstream connections though. I would keep them disabled for initial testing.